azure-pipeline/src/index.ts

```diff
@@ -0,0 +1,113 @@
+import taskLib = require('azure-pipelines-task-lib/task');
+import { AxiosRequestConfig  } from "axios";
+import { updateUserConfig } from './utils/api';
+import {
+    API_VERSION,
+    SCOPE,
+    OUTPUT_PATH,
+    INPUT,
+} from './utils/constants';
+import {
+    FileHandler,
+} from './utils/file-handler';
+import {
+    validatedEnvVar,
+    validatedAuthParams,
+} from './utils/common';
+import { 
+    ClientSecretCredential,
+    AuthenticationError,
+} from '@azure/identity';
+import { info } from 'console';
+
+
+class ConfigUpdater {
+
+    private f = new FileHandler;
+
+    // getAuthorizationTokenFromKey():
+    // a function takes user's input given in the .yml file
+    // and request an bearer token for authentication of later authorization
+    // using @azure/identity
+    private getAuthorizationTokenFromKey = async () => {
+        try {
+            const params = {
+                tenantID:  validatedAuthParams('tenantid'),
+                clientID:  validatedAuthParams('servicePrincipalId'),
+                servicePrincipalKey: validatedAuthParams('servicePrincipalKey'),
+            }
+            const clientSecretCredential: ClientSecretCredential = new ClientSecretCredential(
+                params.tenantID, params.clientID, params.servicePrincipalKey
+            );
+            const accessToken = await clientSecretCredential.getToken(SCOPE);
+            if (!accessToken) {
+                throw new AuthenticationError(0, {
+                    error: 'Access token fetch failed'
+                });
+            }
+            return accessToken;
+        } catch (e) {
+            throw new Error("Authorization parameters validation failed");   
+        }
+    }
+
+    // getConvertedFileObject():
+    // compress config file folder
+    // and make it ready for sending it to backend
+    private getConvertedFileObject = async () => {
+        const file = await this.f.compressFile(
+            validatedEnvVar(INPUT.source),
+            validatedEnvVar(INPUT.target),
+            OUTPUT_PATH,
+        );
+        return {
+            properties: {
+                rootFile: `${validatedEnvVar(INPUT.target)}nginx.conf`,
```

Comment:
This is probably the most common case, but the `rootFile` _must_ be `/some/config/path/nginx.conf`? The file name can't be something else?